tests/test_ble_interface_core.py

```
@@ -6146,6 +6424,111 @@ def start_notify(self, *_args: object, **_kwargs: object) -> None:
+) -> None:
+    """Notification wrappers should use inline fallback when safe_execute hooks are unconfigured."""
+
+    class _ClientWithCallbacks(DummyClient):
+        def __init__(self) -> None:
+            super().__init__()
+            self.callbacks: dict[str, Callable[[Any, bytes], None]] = {}
+
+        def has_characteristic(self, uuid: str) -> bool:
+            return uuid in {LOGRADIO_UUID, FROMNUM_UUID}
+
+        def start_notify(self, *args: Any, **kwargs: Any) -> None:
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)